javacc-wrapper.xml

```diff
@@ -488,17 +488,16 @@ public interface" />
 
     </target>
 
-    <target name="side-effecting-visitor" depends="default-visitor" unless="jjtreeBuildNotRequired">
-        <!-- Side effecting visitor, no return type, one generic parameter -->
-        <copy file="${base-visitor-interface-file}" tofile="${generic-sideeffect-visitor-interface-file}" />
+    <target name="generic-visitor-replacement" depends="default-visitor" unless="jjtreeBuildNotRequired">
+        <move file="${base-visitor-interface-file}" tofile="${generic-sideeffect-visitor-interface-file}" />
         <replace file="${generic-sideeffect-visitor-interface-file}">
             <replacefilter token="${base-visitor-interface-name}"
-                           value="${generic-sideeffect-visitor-interface-name}&lt;T>" />
+                           value="${generic-visitor-interface-name}&lt;P, R> extends ${ast-api-package}.AstVisitor&lt;P, R>" />
 
-            <replacefilter token="Object" value="T" />
-            <replacefilter token="T visit" value="void visit" />
-            <replacefilter token="return data;" value="" />
-            <replacefilter token="return " value="" />
+            <replacefilter token="Object" value="P" />
+            <replacefilter token="P visit" value="R visit" />
+
+            <replacefilter token="default R visit(${node-name} node, P data) { for (int i = 0, len = node.getNumChildren(); i &lt; len; i++) node.getChild(i).jjtAccept(this, data); return data; }" value="default R visit(${node-name} node, P data) { return visitNode(node, data); }" />
```

Comment:
It is correct, because the regular non-generic visitor still uses `jjtAccept`  (see a few lines above). This is to be compatible with other javacc modules until they're ported. When that's done it can be simplified